projects/components/src/download-file/download-file.component.ts

```diff
@@ -1,67 +1,41 @@
-import { DOCUMENT } from '@angular/common';
-import { ChangeDetectionStrategy, ChangeDetectorRef, Component, Inject, Input, Renderer2 } from '@angular/core';
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
 import { IconType } from '@hypertrace/assets-library';
-import { catchError, finalize, take } from 'rxjs/operators';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
 import { ButtonSize, ButtonStyle } from '../button/button';
 import { IconSize } from '../icon/icon-size';
-import { NotificationService } from '../notification/notification.service';
 import { DownloadFileMetadata } from './download-file-metadata';
+import { FileDownloadEventType, FileDownloadService } from './service/file-download.service';
 
 @Component({
   selector: 'ht-download-file',
   changeDetection: ChangeDetectionStrategy.OnPush,
   styleUrls: ['./download-file.component.scss'],
+  providers: [FileDownloadService],
   template: `
     <div *ngIf="this.metadata" class="download-file" (click)="this.triggerDownload()">
       <ht-button
-        *ngIf="!this.dataLoading"
+        *ngIf="!(this.dataLoading$ | async)"
         class="download-button"
         icon="${IconType.Download}"
         display="${ButtonStyle.Text}"
         size="${ButtonSize.Large}"
       ></ht-button>
-      <ht-icon *ngIf="this.dataLoading" icon="${IconType.Loading}" size="${IconSize.Large}"></ht-icon>
+      <ht-icon *ngIf="this.dataLoading$ | async" icon="${IconType.Loading}" size="${IconSize.Large}"></ht-icon>
     </div>
   `
 })
 export class DownloadFileComponent {
   @Input()
   public metadata?: DownloadFileMetadata;
 
-  public dataLoading: boolean = false;
-  private readonly dlJsonAnchorElement: HTMLAnchorElement;
+  public constructor(private readonly fileDownloadService: FileDownloadService) {}
 
-  public constructor(
-    @Inject(DOCUMENT) private readonly document: Document,
-    private readonly renderer: Renderer2,
-    private readonly changeDetector: ChangeDetectorRef,
-    private readonly notificationService: NotificationService
-  ) {
-    this.dlJsonAnchorElement = this.document.createElement('a');
+  public get dataLoading$(): Observable<boolean> {
```

Comment:
Move it to constructor please

Reply:
Not needed, did refactoring after offline discussion